Backpack-SwiftUI/Calendar/Classes/Core/CalendarContainer.swift

```diff
@@ -0,0 +1,75 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+struct CalendarContainer<MonthContent: View>: View {
+    let calendar: Calendar
+    let validRange: ClosedRange<Date>
+    @ViewBuilder let monthContent: (_ month: Date) -> MonthContent
+    
+    var body: some View {
+        ScrollView {
```

Comment:
From what I can tell, you're not doing any cell recycling of the months, I guess that's because there's a limit of 12 months, so it's not necessary, is that correct?

Reply:
the months themselves are not recycled, but the day cells are. if both the day cells and months are dynamically resolved while scrolling, that produces jumps in the UI while scrolling, because a new month will load but then more cells need to render (the days), pushing content up more.

Later on, when we can support using the Grid component with iOS 16 (only LazyVGrid is available for ios 14 :/ ) we'll be able to switch to recycling months and not the days, which would be preferred!

So we're kinda cornered here by the ios versions we support!